hack/set-motd/main.go

```diff
@@ -0,0 +1,165 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"time"
+)
+
+type stringFlag struct {
+	set   bool
+	value string
+}
+
+func (sf *stringFlag) Set(x string) error {
+	sf.value = x
+	sf.set = true
+	return nil
+}
+
+func (sf *stringFlag) String() string {
```

Comment:
String() could make confusion with the standard String function...should we use something like getValue() for example

Reply:
whatever you want instead of reserved name for a existing name in the standard packages

Reply:
I forgot one key detail, Set and String are functions implemented so the interface used by stringFlag can be used, which in the end is interacting with the flag pkg.

So String can't be renamed and actually Set *has* to return nil.

Reply:
whatttttt??? check it out please you're just only using flag package as a import, the issue could be that you select the same name than a reserved type...

Anyway ... for me it's LGTM just for internal use

Reply:
I think you're missing a point about using the type flag.Value here Alberto, just did a new test for showing/learning purposes:

- Added some debug code:
```
	fmt.Printf("motd: %t, %s\n", motd.set, motd.value)
	fmt.Printf("user: %t, %s\n", user.set, user.value)
	fmt.Printf("pr: %t, %s\n", pr.set, pr.value)
	fmt.Printf("customPath: %t, %s\n", customPath.set, customPath.value)
```

- With the initial Set and String codes, the output looks like:
```
set-motd]$ go run main.go set -motd MOTD-MSG -path $(pwd)/ocatopic -pr 1234

motd: true, MOTD-MSG
user: false, ZTPFW Github Actions
pr: true, 1234
customPath: true, /home/aranda/projects/ztp-pipeline-relocatable/hack/set-motd/ocatopic
```

- Changing the Set function to not return nil would change the interface contract, hence not implementing it. So to test this theory I changed it to not return the error:
```
func (sf *stringFlag) Set(x string) {
	sf.value = x
	sf.set = true
}
```

- Then, the output shows that the type is not used correctly (because we don't implement anymore the interface):
```
]$ go run main.go set -motd MOTD-MSG -path $(pwd)/ocatopic -pr 1234

# command-line-arguments
./main.go:149:13: cannot use &setVarMotd (type *stringFlag) as type flag.Value in argument to setCmd.Var:
	*stringFlag does not implement flag.Value (wrong type for Set method)
		have Set(string)
		want Set(string) error
./main.go:150:13: cannot use &setVarUser (type *stringFlag) as type flag.Value in argument to setCmd.Var:
	*stringFlag does not implement flag.Value (wrong type for Set method)
		have Set(string)
		want Set(string) error
./main.go:151:13: cannot use &setVarPR (type *stringFlag) as type flag.Value in argument to setCmd.Var:
	*stringFlag does not implement flag.Value (wrong type for Set method)
		have Set(string)
		want Set(string) error
./main.go:152:13: cannot use &setCustomPath (type *stringFlag) as type flag.Value in argument to setCmd.Var:
	*stringFlag does not implement flag.Value (wrong type for Set method)
		have Set(string)
		want Set(string) error
```

The Set is called by setCmd.Var, and it expects a Set and a String function according to the type Var, which we implement using our custom type stringFlag. Named that way because we are only using strings, on the other hand.

Regarding the string function `func (sf *stringFlag) String() string` it doesn't collide with any other string function in go as it just expects our custom type.